internal/service/infra_mgmt/api_hosts_resource.go

```diff
@@ -0,0 +1,168 @@
+	var data InfraHostModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, _, err := r.client.InfraManagementAPI.
+		HostsAPI.
+		HostsCreate(ctx).
+		Body(*data.Expand(ctx, &resp.Diagnostics)).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to create Hosts, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResult()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *HostsResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data InfraHostModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, httpRes, err := r.client.InfraManagementAPI.
+		HostsAPI.
+		HostsRead(ctx, data.Id.ValueString()).
+		Execute()
+	if err != nil {
+		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read Hosts, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResult()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *HostsResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data InfraHostModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	apiRes, _, err := r.client.InfraManagementAPI.
+		HostsAPI.
+		HostsUpdate(ctx, data.Id.ValueString()).
+		Body(*data.Expand(ctx, &resp.Diagnostics)).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to update Hosts, got error: %s", err))
+		return
+	}
+
+	res := apiRes.GetResult()
+	data.Flatten(ctx, &res, &resp.Diagnostics)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *HostsResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data InfraHostModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	httpRes, err := r.client.InfraManagementAPI.
+		HostsAPI.
+		HostsDelete(ctx, data.Id.ValueString()).
+		Execute()
+	if err != nil {
+		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+			return
+		}
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to delete Hosts, got error: %s", err))
+		return
```

Comment:
Why not return the error if 404?

Reply:
It is to handle the case if a resource was deleted externally